### DISCUSSION TITLE
auto code changes based on spotbug report

### DISCUSSION THREAD

**Discussion by mahmedhm** [spotbugs/spotbugs#2100] [General]
We have a huge code base and we are thinking about enabling soft bugs. I am wondering if there are tools available that can make those changes automatically and all we have to do is review it manually.

**Comment by ThrawnCA** [comment]
SpotBugs doesn't have any features like that. I'm not sure what tools do exist, sorry.

**Reply by mahmedhm** [reply]
thanks for the prompt response... appreciate it.

**Comment by KengoTODA** [comment]
You may try to use https://errorprone.info/docs/patching instead of SpotBugs.

**Reply by KengoTODA** [reply]
For Kotlin, https://detekt.dev/ provides auto-correct in some rules.
